### DISCUSSION TITLE
How can I use the probability for positive class in my new metric?

### DISCUSSION THREAD

**Discussion by hadar-ver** [pycaret/pycaret#3706] [General]
Hi,
I would like to add metric for my binary classification problem of predict the probability for insurance event. I will use custom 'gini coefficient' metric.
As far as I know, the add_metric() function can get only 2 parameters: y_true and y_pred; y_pred can be either predicted label or predicted probability. The problem is that the predicted probability is for the probability to be in the predicted class and it doesn't meet my needs to get the probability to be in negative class.

So I have 2 questions that will help me to have solution to my problem:
1. Can I pass to my custom metric 3 arguments from the model (y_true, y_pred_label, y_pred_score)?
2. Or can I pass to my custom metric the predicted probability to be in positive class?

This is an example of predicted labels 0 with prediction scores. I would like to get the (1-prediction scores) for predicted class 0 and prediction scores for predicted class 1: 
![image](https://github.com/pycaret/pycaret/assets/71387302/5b0c2dc4-5ca3-4018-a148-5065331e685a)

Thanks in advance for any help :)

**Comment by celestinoxp** [comment]
@ngupta23 Do you want to respond?

**Comment by moezali1** [comment]
@hadar-ver If you can design a function that does exactly what you explain and pass it in `score_func` in `add_metric`, it should work. 

Read the documentation here: https://pycaret.readthedocs.io/en/latest/api/classification.html#pycaret.classification.add_metric

**Reply by hadar-ver** [reply]
Hi, thanks for replying :)
In the documentation I can see the following: _Score function (or loss function) with signature score_func(y, y_pred, **kwargs)._
However, my function takes: def custom(y_true, predicted_class, probability)

How can I pass y_true, predicted_class and probability to the score functions that takes only y,y_pred, **kwargs?
If there is a way to do so, it will be really helpful for me if you could write an example :)

Thanks!
